src/deviceinfo.cpp

```diff
@@ -51,9 +52,11 @@ class DeviceInfoPrivate
     QString m_osName;
     QString m_osVersion;
     QString m_adaptationVersion;
+    OFonoTracker *m_ofonoTracker;
 };
 
-DeviceInfoPrivate::DeviceInfoPrivate()
+DeviceInfoPrivate::DeviceInfoPrivate(bool synchronousInit)
+    : m_ofonoTracker(new OFonoTracker(synchronousInit, nullptr))
```

Comment:
Separate class, having ofono tracking on demand, as singleton, etc would be things to consider if/when the impl is otherwise not bonkers...

Reply:
By the looks of it, libqofono that @monich brought up already has all the low level technical details covered (shared instances, optional blocking, whatnot). I wish I would have known about it a bit earlier... having a new parallel implementation of essentially the same thing with possibility for fresh set of hiccups probably does not make too much sense.

Now, does it also provide a way to "get all imeis" that is easy enough that we can use it as is - without any glue classes? 

If not, having a wrapper class that != "DeviceInfo" using libqofono would probably be preferable - any suggestions about naming?

Reply:
Custom ofono tracking dropped in favor of using libqofono. @monich please check if it looks sane.

Regarding permissions: a process that does not have them, gets empty imei set. Which sounds about right to me. Or at least better than splitting things into multiple interfaces based on current permission requirements, or?

Also changed things so that D-Bus access is done on demand so that no ofono access is done unless imei numbers are queried -> we can still move this to somewhere else, but that would be arch decision rather than technical one.